prog.go

```diff
@@ -361,6 +361,13 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions, handles *hand
 		}
 	}
 
+	if (errors.Is(logErr, unix.EINVAL) || errors.Is(logErr, unix.EPERM)) &&
```

Comment:
I think `EINVAL` happens a lot, so we'll do another traversal of all instructions for most errors. Plus we'll allocate a slice just to get it's length. What if you return `hasB2BCalls bool` from `fixupAndValidate` or similar and gate on that here instead? Alternatively, a function that returns on the first found call would also be an improvement.